ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/PullProcessRepository.scala

```diff
@@ -1,19 +1,28 @@
 package pl.touk.nussknacker.ui.process.repository
 
-import pl.touk.nussknacker.restmodel.processdetails.BaseProcessDetails
+import pl.touk.nussknacker.engine.api.process.ProcessName
+import pl.touk.nussknacker.restmodel.processdetails.{BaseProcessDetails, ProcessShapeFetchStrategy}
 import pl.touk.nussknacker.restmodel.process
+import pl.touk.nussknacker.restmodel.process.ProcessId
 import pl.touk.nussknacker.ui.listener.services.{PullProcessRepository => ListenerPullProcessRepository}
 import pl.touk.nussknacker.ui.security.api.LoggedUser
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class PullProcessRepository(fetchingProcessRepository: FetchingProcessRepository[Future]) extends ListenerPullProcessRepository {
 
-  override def fetchLatestProcessDetailsForProcessId(id: process.ProcessId)
-                                                    (implicit loggedUser: LoggedUser, ec: ExecutionContext): Future[Option[BaseProcessDetails[Unit]]] =
+  override def fetchLatestProcessDetailsForProcessId[PS: ProcessShapeFetchStrategy](id: process.ProcessId)
+                                                                                   (implicit loggedUser: LoggedUser, ec: ExecutionContext): Future[Option[BaseProcessDetails[PS]]] =
     fetchingProcessRepository.fetchLatestProcessDetailsForProcessId(id = id, businessView = false)
 
-  override def fetchProcessDetailsForId(processId: process.ProcessId, versionId: Long)
-                                       (implicit loggedUser: LoggedUser, ec: ExecutionContext): Future[Option[BaseProcessDetails[Unit]]] =
+  override def fetchProcessDetailsForId[PS: ProcessShapeFetchStrategy](processId: process.ProcessId, versionId: Long)
+                                                                      (implicit loggedUser: LoggedUser, ec: ExecutionContext): Future[Option[BaseProcessDetails[PS]]] =
     fetchingProcessRepository.fetchProcessDetailsForId(processId, versionId, businessView = false)
+
+  override def fetchProcessDetailsForName[PS: ProcessShapeFetchStrategy](processName: ProcessName, versionId: Long)
+                                                                        (implicit loggedUser: LoggedUser, ec: ExecutionContext): Future[Option[BaseProcessDetails[PS]]] = for {
+    maybeProcessId <- fetchingProcessRepository.fetchProcessId(processName)
+    processId <- maybeProcessId.fold(Future.failed[ProcessId](new RuntimeException(s"ProcessId for $processName not found")))(Future.successful)
```

Comment:
Do we have some better exception for missing data on this layer? If not, maybe better will be at lease `IllegalArgumentException`?

Reply:
Ok, let's throw `IllegalArgumentException`